src/test/java/org/snomed/snowstorm/core/data/services/BranchMergeServiceTest.java

```diff
@@ -3630,6 +3630,405 @@ void assertReleaseProperties(
+		branchService.create(extTaskA);
+
+		// 10. Extension changes hierarchy of Car through classification, i.e. it is now top-level
+		concept = conceptService.find(carId, extTaskA);
+		concept.getClassAxioms().iterator().next().setActive(false);
+		concept.getRelationships().iterator().next().setActive(false);
+		concept.getClassAxioms().iterator().next().setRelationships(Set.of(new Relationship(ISA, SNOMEDCT_ROOT)));
+		concept.getRelationships().add(new Relationship(ISA, SNOMEDCT_ROOT));
+		conceptService.update(concept, extTaskA);
+
+		// 11. Promote task A
+		branchMergeService.mergeBranchSync(extTaskA, extProject, Collections.emptySet());
+
+		// 12. In International, inactivate association member
+		// Author decides to inactivate the member as strictly speaking plural does not necessarily mean SAME_AS
+		String associationMemberId = associationMember.getMemberId();
+		associationMember = memberService.findMember(intMain, associationMemberId);
+		associationMember.setActive(false).updateEffectiveTime();
+		memberService.updateMember(intMain, associationMember);
+
+		// 13. Version International
+		codeSystem = codeSystemService.find("SNOMEDCT");
+		codeSystemService.createVersion(codeSystem, 20220331, "20220331");
+
+		// 14. Upgrade Extension
+		codeSystem = codeSystemService.find("SNOMEDCT-TEST");
+		codeSystemUpgradeService.upgrade(codeSystem, 20220331, false);
+
+		// 15. Assert state of inactivated member before rebase
+		associationMember = memberService.findMember(intMain, associationMemberId);
+		assertEquals(CORE_MODULE, associationMember.getModuleId());
+		assertVersioned(associationMember, 20220331); // International has latest version
+
+		associationMember = memberService.findMember(extProject, associationMemberId);
+		assertEquals(CORE_MODULE, associationMember.getModuleId());
+		assertVersioned(associationMember, 20220228); // Not yet rebased latest version
+
+		// 16. Rebase Extension project
+		review = getMergeReviewInCurrentState(extMain, extProject);
+		conflicts = reviewService.getMergeReviewConflictingConcepts(review.getId(), new ArrayList<>());
+		assertTrue(conflicts.size() != 0);
+		// Doesn't matter which column is selected
+		for (MergeReviewConceptVersions conflict : conflicts) {
+			reviewService.persistManuallyMergedConcept(review, Long.parseLong(carId), conflict.getAutoMergedConcept());
+		}
+		reviewService.applyMergeReview(review);
+
+		// 17. Assert state of inactivated member after rebase
+		associationMember = memberService.findMember(intMain, associationMemberId);
+		assertEquals(CORE_MODULE, associationMember.getModuleId());
+		assertVersioned(associationMember, 20220331);
+
+		associationMember = memberService.findMember(extProject, associationMemberId);
+		assertEquals(CORE_MODULE, associationMember.getModuleId());
+		assertVersioned(associationMember, 20220331);
+	}
+
+	@Test
+	void testPromotionDoesNotResultInExtensionStealingAssociationMember() throws ServiceException, InterruptedException {
```

Comment:
This test shows the inactive historical association does not switch module when promoting.